### PR TITLE
Issues enhancements

### PR DESCRIPTION
**Pull request by veggiematts**
This patch adds several enhancements to issues tracking is the organizations module : 

- A type can be added to issues. Those issues types can be defined in the admin page.
- Issues can now have a start and an end date.
- Issues can be exported as CSV, either all the issues or per-organization issues only.

issues_types_admin
Issues types definition in the admin page.

issues_add
Adding a new issue with the new features: start date, end date and type of issue.

issues_list
A list of issues for an organization, with the new features, and with the CSV export options.

### CODE REVIEW

**Comment by jeffnm** [comment]
I've installed this on our development server to do some local testing.

**Comment by jeffnm** [comment]
It's been more than 2 weeks on this.  I'm still waiting on some testing from others locally to verify this is working as expected, but it looks good to me. If anyone else has been able to test this change, then we can probably go ahead and merge it.

**Comment by benheet** [comment]
Hi Jeffrey, I'm not sure we'll have others with time to test this. We should be ok with your review and opinion this look good. I think you can go ahead and merge it.

**Comment by PaulPoulain** [comment]
Hi Matts,

either I'm blind, you you haven't explained how to update the database (and I can't find any update.sql file as well)

**Comment by veggiematts** [comment]
The upgrade file has been pushed : install/upgrade_issues_enhancements.sql
You can execute it as it is to update the database.

File: ajax_forms.php [hunk 1]
```diff
@@ -524,12 +524,23 @@
...
+			$issueStartDate='';
 		}
 
+    if (($issueLog->issueEndDate != '') && ($issueLog->issueEndDate != "0000-00-00")) {
+			$issueEndDate=format_date($issueLog->issueEndDate);
+		}else{
+			$issueEndDate='';
+		}
+
+    $issueLogArray =
```
Comment (remocrevo) [inline ajax_forms.php]: It looks like this line is not finished.

File: admin/classes/domain/IssueLog.php [hunk 2]
```diff
@@ -23,7 +23,38 @@ protected function defineRelationships() {}
...
+  }
+
+  public function allExpandedAsArray($organizationID) {
+
+    $query = "SELECT IssueLog.*, Organization.name, IssueLogType.shortName from IssueLog
+          LEFT JOIN IssueLogType ON IssueLogType.issueLogTypeID = IssueLog.issueLogTypeID
+          LEFT JOIN Organization ON Organization.organizationID = IssueLog.organizationID";
+
+    if ($organizationID) {
+      $query .= " WHERE Organization.OrganizationID = $organizationID";
```
Comment (remocrevo) [inline admin/classes/domain/IssueLog.php]: When including parameters in SQL queries, we should sanitize the input using something like mysqli_real_escape_string().

**Comment by jeffnm** [comment]
I've run the upgrade_issues_enhancements.sql with no errors.

**Comment by remocrevo** [comment]
Thanks @jeffnm for testing the upgrade script. After @veggiematts addresses the small code issues I mentioned above, I think this will be ready for merging.

**Comment by veggiematts** [comment]
- Remove incomplete instruction.
  - Fix escapeString function.
  - Use escapeString function in allExpandedAsArray.

We cannot use mysqli_real_escape_string because Coral is currently using mysql, not mysqli.
As the escapeString function was not used anywhere yet, I've modified it to use mysql_real_escape_string, so the function is now usable.

**Comment by remocrevo** [comment]
Thanks @veggiematts for fixing those problems.  We will need to update CORAL to use mysqli eventually, but that is a separate project.  So this pull request is ready to merge.  For the record, @jeffnm also tested the modified database install script without any errors.

**Comment by benheet** [comment]
This is a great new feature. Matthias and/or Paul P, can you write up an announcement for these new features and send it out to the coral community listserv?

**Comment by remocrevo** [comment]
Before we announce to the email list, we should also make sure we have updated install and upgrade instructions in the appropriate places (README? User guide?) and use GitHub to create a new release with a new version number. I can do the release.  Can others help with the documentation?

**Comment by benheet** [comment]
Matthias, I think we'll need you to update and supply the install and upgrade instructions.

**Comment by veggiematts** [comment]
Okay, what are the appropriate places to do so ?

**Comment by remocrevo** [comment]
I think the best place right now is on the CORAL-Main wiki.  Here is the existing page: https://github.com/ndlibersa/CORAL-Main/wiki/CORAL-Resources-User-Guide

**Comment by benheet** [comment]
what was the result of testing whether we could get images embedded in the page?

**Comment by remocrevo** [comment]
I propose we add wiki images to the code repository for CORAL-Main, where the wiki is.  Once an image is in the repository, you can get its URL by viewing the image in GitHub, clicking the "Raw" button, and copying the URL from the address bar.  Use that URL when adding an image to the wiki.  If you want to create the URL manually, this is the pattern:

https://github.com/ndlibersa/CORAL-Main/raw/master/images/butterflyfishlogo.gif

**Comment by remocrevo** [comment]
The downside of this approach is that each CORAL installation includes all those images without the wiki text.  The wiki actually has its own repository, which would keep the images separate from the CORAL installation, but it doesn't have a nice GitHub interface; you have to clone it locally to add the images.

**Comment by veggiematts** [comment]
Has something been decided regarding the images in the documentation ? 

The problem with using the Coral-Main repository for storing the images is that everybody is not allowed to push to it, when the wiki is editable by anyone.

**Comment by PaulPoulain** [comment]
Le 15/04/2015 15:40, Matthias Meusburger a écrit :
>
> Has something been decided regarding the images in the documentation ?
>
> The problem with using the Coral-Main repository for storing the 
> images is that everybody is not allowed to push to it, when the wiki 
> is editable by anyone.
>
>
A solution could be to have a "documentation image" repository, that is 
more widely open to push.

-- 
Paul Poulain, Associé-gérant / co-owner
BibLibre, Services en logiciels libres pour les bibliothèques
BibLibre, Open Source software and services for libraries

**Comment by veggiematts** [comment]
I agree with this idea.

**Comment by remocrevo** [comment]
I think it's time to discuss documentation more generally. Ben recently sent an email to the Admin list suggesting one option (http://docs.readthedocs.org/), and other options could also work for us. But if we decide that we need a separate repository for our doc images, I think it makes sense to also have the doc files stored there.

**Pull request merged**